delft3dworker/views.py

```diff
@@ -407,6 +410,22 @@ def publish_company(self, request, pk=None):
 
         return Response({'status': 'Published scene to company'})
 
+    @list_route(methods=["post"])  # denied after publish to world
+    def publish_company_all(self, request):
+        queryset = Scene.objects.filter(owner=self.request.user).filter(
+                suid__in=request.data.getlist('suid', []))
+
+        try:
+            for scene in queryset:
+                scene.publish_company(request.user)
+        except ValueError, e:
```

Comment:
https://docs.djangoproject.com/en/1.10/topics/db/queries/#querysets-are-lazy

Which means that the "`suid` correctness check" happens on r.419.

(The long read: https://docs.djangoproject.com/en/1.10/ref/models/querysets/#when-querysets-are-evaluated)